### ISSUE TITLE
TLS doesn't work with grpc when using linkerd from Docker images

### ISSUE THREAD

**Issue by pinak** [BuoyantIO/linkerd#861]
I'm using linkerD docker image (v 0.8.4) from https://hub.docker.com/r/buoyantio/linkerd/ and when I try using it with gRpc and TLS I get: java.lang.UnsupportedOperationException: JDK provider does not support NPN_AND_ALPN protocol

and when I try using it with OpenJdk 9 it seems to work.

**Comment by olix0r** [comment]
I'm pretty sure I know what's going on here:

```
:; docker run -it --rm --entrypoint=bash buoyantio/linkerd:0.8.4 
root@3306c8d614d7:/io.buoyant/linkerd/0.8.4# env
HOSTNAME=3306c8d614d7
TERM=xterm
L5D_EXEC=/io.buoyant/linkerd/0.8.4/bundle-exec
CA_CERTIFICATES_JAVA_VERSION=20140324
L5D_HOME=/io.buoyant/linkerd/0.8.4
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/io.buoyant/linkerd/0.8.4
JAVA_HOME=/usr/lib/jvm/java-8-openjdk-i386
LANG=C.UTF-8
JAVA_VERSION=8u91
SHLVL=1
HOME=/root
JAVA_DEBIAN_VERSION=8u91-b14-1~bpo8+1
no_proxy=*.local, 169.254/16
_=/usr/bin/env
```

We're trying to use the 32b JVM image (which uses [less memory](https://blog.buoyant.io/2016/06/17/small-memory-jvm-techniques-for-microservice-sidecars/)):
> JAVA_HOME=/usr/lib/jvm/java-8-openjdk-**i386**

However, the boringssl fatjar shipped with linkerd is surely only compiled against the 64b jvm.

We probably need to use a 64b jvm by default and allow users to move to 32b jvms as needed.

**Comment by olix0r** [comment]
Running with `JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64` fixes the issue for me, which confirms the above.

**Comment by olix0r** [comment]
In the latest release, we changed the default docker image to depend on library/java, which resolves this issue. (32b images are published separately)

**Comment by wmorgan** [comment]
@pinak if you have a moment to confirm the fix with 0.8.5, that would be great!
